lib/Parse/Lexer.cpp

```diff
@@ -2198,24 +2196,22 @@ void Lexer::lexImpl() {
       return lexOperatorIdentifier();
 
     bool ShouldTokenize = lexUnknown(/*EmitDiagnosticsIfToken=*/true);
-    if (ShouldTokenize) {
-      return formToken(tok::unknown, TokStart);
-    }
-    goto Restart; // Skip presumed whitespace.
+    assert(
+        ShouldTokenize &&
+        "Invalid UTF-8 sequence should be eaten by lexTrivia as LeadingTrivia");
```

Comment:
This may cause unused variable warning in `--no-assertion` build. Please add `(void)ShouldTokenize;` line here.